server/controllers/finance/reports/ohada_balance_sheet/index.js

```diff
@@ -0,0 +1,299 @@
+  orientation : 'landscape',
+  footerRight : '[page] / [toPage]',
+};
+
+const balanceSheetAssetTable = [
+  { ref : 'AD', is_title : 1 },
+  { ref : 'AE', is_title : 0 },
+  { ref : 'AF', is_title : 0 },
+  { ref : 'AG', is_title : 0 },
+  { ref : 'AH', is_title : 0 },
+  { ref : 'AI', is_title : 1 },
+  { ref : 'AJ', is_title : 0 },
+  { ref : 'AK', is_title : 0 },
+  { ref : 'AL', is_title : 0 },
+  { ref : 'AM', is_title : 0 },
+  { ref : 'AN', is_title : 0 },
+  { ref : 'AP', is_title : 0 },
+  { ref : 'AQ', is_title : 1 },
+  { ref : 'AR', is_title : 0 },
+  { ref : 'AS', is_title : 0 },
+  { ref : 'AZ', is_title : 1 },
+  { ref : 'BA', is_title : 0 },
+  { ref : 'BB', is_title : 0 },
+  { ref : 'BG', is_title : 0 },
+  { ref : 'BH', is_title : 0 },
+  { ref : 'BI', is_title : 0 },
+  { ref : 'BJ', is_title : 0 },
+  { ref : 'BK', is_title : 1 },
+  { ref : 'BQ', is_title : 0 },
+  { ref : 'BR', is_title : 0 },
+  { ref : 'BS', is_title : 0 },
+  { ref : 'BT', is_title : 1 },
+  { ref : 'BU', is_title : 0 },
+  { ref : 'BZ', is_title : 1 },
+];
+
+const balanceSheetLiabilityTable = [
+  { ref : 'CA', is_title : 1 },
+  { ref : 'CB', is_title : 0 },
+  { ref : 'CC', is_title : 0 },
+  { ref : 'CD', is_title : 0 },
+  { ref : 'CE', is_title : 0 },
+  { ref : 'CF', is_title : 0 },
+  { ref : 'CG', is_title : 0 },
+  { ref : 'CH', is_title : 0 },
+  { ref : 'CI', is_title : 1 },
+  { ref : 'CK', is_title : 1 },
+  { ref : 'CL', is_title : 0 },
+  { ref : 'CM', is_title : 0 },
+  { ref : 'CP', is_title : 1 },
+  { ref : 'DA', is_title : 0 },
+  { ref : 'DB', is_title : 0 },
+  { ref : 'DC', is_title : 0 },
+  { ref : 'DD', is_title : 0 },
+  { ref : 'DF', is_title : 1 },
+  { ref : 'DG', is_title : 1 },
+  { ref : 'DH', is_title : 0 },
+  { ref : 'DI', is_title : 0 },
+  { ref : 'DJ', is_title : 0 },
+  { ref : 'DK', is_title : 0 },
+  { ref : 'DM', is_title : 0 },
+  { ref : 'DN', is_title : 0 },
+  { ref : 'DP', is_title : 0 },
+  { ref : 'DQ', is_title : 0 },
+  { ref : 'DR', is_title : 0 },
+  { ref : 'DS', is_title : 0 },
+  { ref : 'DT', is_title : 1 },
+  { ref : 'DV', is_title : 1 },
+  { ref : 'DZ', is_title : 1 },
+];
+
+// expose to the API
+exports.document = document;
+
+/**
+ * @function document
+ * @description process and render the balance report document
+ */
+function document(req, res, next) {
+  const params = req.query;
+  const context = {};
+  let report;
+
+  _.defaults(params, DEFAULT_PARAMS);
+
+  try {
+    report = new ReportManager(TEMPLATE, req.session, params);
+  } catch (e) {
+    next(e);
+    return;
+  }
+
+  getFiscalYearDetails(params.fiscal_id)
+    .then(fiscalYear => {
+      _.merge(context, { fiscalYear });
+
+      const dbPromises = [];
+      const usePrevious = fiscalYear.previous.period_id ?
+        AccountReference.computeAllAccountReference(fiscalYear.previous.period_id) : [];
+
+      dbPromises.push(AccountReference.computeAllAccountReference(fiscalYear.current.period_id));
```

Comment:
Do you need to make a `dbPromises` array here?  It disguises the fact there there are only two elements.

I propose you just give both results a name and put them in the array one at a time.  It will save you a line of code as well:
```js
const currentPeriodReferences = AccountReference.computeAllAccountReference(fiscalYear.current.period_id);
const previousPeriodReferences = fiscalYear.previous.period_id ?
        AccountReference.computeAllAccountReference(fiscalYear.previous.period_id) : [];

return Q.all([currentPeriodReferences, previousPeriodReferences]);
```

In my mind it makes it a bit more clear.